internal/routing/handlers/user_handler.go

```diff
@@ -216,6 +219,123 @@ func (handler *UserHandler) ResendToken(w http.ResponseWriter, r *http.Request)
+	if _, err := tx.Exec(transactionCtx, queryString, nicknameChangeRequest.NewNickname, userID); err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Retrieve the updated user
+	userDto := &schemas.UserDTO{}
+	queryString = "SELECT username, nickname, email FROM alpha_schema.users WHERE user_id = $1"
+	if err := tx.QueryRow(transactionCtx, queryString, userID).Scan(&userDto.Username, &userDto.Nickname, &userDto.Email); err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	if err := utils.CommitTransaction(w, tx, transactionCtx, cancel); err != nil {
+		return
+	}
+
+	// Send the updated user in the response
+	w.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(w).Encode(userDto); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+}
+
+func (handler *UserHandler) ChangePassword(w http.ResponseWriter, r *http.Request) {
+	// Begin a new transaction
+	tx, transactionCtx, cancel := utils.BeginTransaction(w, r, handler.DatabaseManager.GetPool())
+	if tx == nil || transactionCtx == nil {
+		return
+	}
+	var err error
+	defer utils.RollbackTransaction(w, tx, transactionCtx, cancel, err)
+
+	// Decode the request body into the password change request struct
+	passwordChangeRequest := &schemas.PasswordChangeRequest{}
+	if err = utils.DecodeRequestBody(w, r, passwordChangeRequest); err != nil {
+		return
+	}
+
+	// Get username from path
+	username := chi.URLParam(r, "username")
+
+	// Validate the password change request struct using the validator
+	if err = utils.ValidateStruct(w, passwordChangeRequest); err != nil {
+		return
+	}
+
+	// Get the user ID
+	_, userID, errorOccurred := retrieveUserIdAndEmail(transactionCtx, w, tx, username)
+	if errorOccurred {
+		return
+	}
+
+	//Check if old password is correct
+	if err = checkPassword(w, transactionCtx, tx, username, passwordChangeRequest.OldPassword); err != nil {
+		return
+	}
+
+	// Hash the new password
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(passwordChangeRequest.NewPassword), bcrypt.DefaultCost)
+	if err != nil {
+		utils.WriteAndLogError(w, schemas.InternalServerError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Update the user's password in the database
+	queryString := "UPDATE alpha_schema.users SET password = $1 WHERE user_id = $2"
+	if _, err = tx.Exec(transactionCtx, queryString, hashedPassword, userID); err != nil {
+		utils.WriteAndLogError(w, schemas.DatabaseError, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Commit the transaction
+	if err = utils.CommitTransaction(w, tx, transactionCtx, cancel); err != nil {
+		return
+	}
+
+	// Send success response
+	w.WriteHeader(http.StatusOK)
+	if err = json.NewEncoder(w).Encode(map[string]string{"message": "Password updated successfully"}); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
```

Comment:
Einmal die Reihenfolge ändern bitte. Zuerst den Body in die Response encoden und dann den `StatusOK`-Header setzen.